docs/assumptions-and-invariants.md

```diff
@@ -0,0 +1,6 @@
+# When to use ASSUME/ASSUMPTION and when to use INVARIANT/INVARIANTS with TLC?
+
+The `ASSUME` (or `ASSUMPTION`) construct in TLA+ is used to introduce constant-level (level 0) formulas—facts that must hold about the constants and fixed structures of a specification. These constraints are evaluated before TLC begins exploring behaviors and are not interpreted as properties of system states.
+A constant-level (level 0) formula contains no state variables, no primed variables, and no temporal operators. It expresses purely mathematical truths about constants, sets, or other static objects. Such formulas might assert, for example, that a constant N is a natural number or that a given set S is nonempty. These belong in ASSUME clauses within the module.
```

Comment:
Here's my attempt at explaining the same:

> The `ASSUME` (or `ASSUMPTION`) construct in TLA+ is used to introduce constant-level (level 0) formulas (formulas that don't contain variables) as constraints. TLC checks all `ASSUME` constraints before generating initial states.
>
> `ASSUME` is mainly used to constrain the possible values of constants, for example to ensure that `NumThreads > 1` or `Readers \intersect Writers = {}`.

I also think it's worth pointing out that you can put `TLCSet` and `PrintT` in ASSUME statements but not sure how to best communicate that

Reply:
I believe mentioning `TLCSet` and `PrintT` is beyond the scope of this help file. However, I anticipate that we will eventually have links between help files and pointers to background reading material.